### DISCUSSION TITLE
AxiosError: Request failed with status code 400,

### DISCUSSION THREAD

**Discussion by bskou57** [ansibleguy76/ansibleforms#215] [Q&A] [answered]
Hi

I just add awx connection with user/password and I get below error when I click "Test Awx"

AxiosError: Request failed with status code 400,

I am running version 5.0.4 of ansibleforms (to fix issue with use_credential)

Thanks for your support

**Comment by bskou57** [comment]
Hi

I do not have dns/firewall issue

Note that I already configured without issue ldap connection (from AD) and also github connection from my private repo

I can successfully ping and nc https port of awx server from docker container

Note that I am using custom https certificate on our aansible awx server

Thanks for your help

**Comment by bskou57** [comment]
A HTTP 400 status code usually means that the request was malformed or invalid, and the server could not understand it. The error could be due to a wrong URL, a missing or incorrect parameter, or an authorization issue. To fix this error, you would need to check the request and response objects to identify what went wrong, and correct the issue.

**Comment by ansibleguy76** [comment]
I simply can't reproduce this.  can you curl from within the container and see the http response ?
what do the logs for AF say ?  You can get them and see them through the help menu -> logs.  Or you can do "docker logs"

**Comment by bskou57** [comment]
Hello

works fine 👍 /app # ping tower.cgroup.com
PING tower.cedelgroup.com (10.16.13.11): 56 data bytes
64 bytes from 10.16.13.11: seq=0 ttl=63 time=0.283 ms
^C
--- tower.cedelgroup.com ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 0.283/0.283/0.283 ms
/app # nc -vvv tower.cgroup.com 443
tower.cedelgroup.com (10.16.13.11:443) open

**Comment by bskou57** [comment]
2024-08-21 15:42:01:421 info: Loading /app/dist/persistent/forms.yaml
2024-08-21 15:42:01:421 info: merging file hello_world.yml
2024-08-21 15:42:01:421 info: merging file maintenance.yaml
2024-08-21 15:42:01:421 notice: Launching form HelloWorld
2024-08-21 15:42:01:421 notice: Creating job
2024-08-21 15:42:01:421 info: searching job template HelloWorld
2024-08-21 15:42:01:421 notice: Updating job 113 failed
2024-08-21 15:42:01:421 notice: Sending status mail for jobid 113
2024-08-21 15:42:01:421 notice: Finding jobid 113
2024-08-21 15:42:01:421 info: Finding job 113
2024-08-21 15:42:01:421 notice: No notifications set
2024-08-21 15:42:01:421 error: Errors :  failed to launch awx template HelloWorld
Request failed with status code 400
2024-08-21 15:42:03:423 info: Finding job 113
2024-08-21 15:42:05:425 info: Finding job 113

**Comment by ansibleguy76** [comment]
Hi,

This log output is not coming from the "test connection".  You are launching the helloworld test form.  Which targets the template HelloWorld => if you don't have that template, it makes sense you get a 400 error.  But the awx test connection is simply loading all templates as test.  That should work fine.  Can you verify ?

Can you also do a CTRL-F5 (after an upgrade it happens that old client code is cached)

**Comment by ansibleguy76** [comment, marked as answer]
![image](https://github.com/user-attachments/assets/672cf9b7-c734-4b6e-8add-6bd0eab95a9a)

do you see that test button ?  Does that work ?

**Comment by ansibleguy76** [comment]
If all is clear and fixed, can you let me know ?

**Comment by ansibleguy76** [comment]
stale.... will assume fixed

**Comment by bskou57** [comment]
Hello

When I click on Test for awx connection I have below error 👍 👍 
AxiosError: Request failed with status code 400

But nothing in "job log"

I did ctrl-f5 without success

Best regards

**Reply by ansibleguy76** [reply]
try curl in the container.  i can't say it is a bug. you must do something wrong. try token and basic authentication.
